arangod/VocBase/vocbase.cpp

```diff
@@ -81,6 +97,108 @@
 using namespace arangodb;
 using namespace arangodb::basics;
 
+struct arangodb::VocBaseLogManager {
+  explicit VocBaseLogManager(application_features::ApplicationServer& server, DatabaseID database)
+      : _server(server),
+        _logContext(LoggerContext{Logger::REPLICATION2}.with<logContextKeyDatabaseName>(
+            std::move(database))) {}
+
+  [[nodiscard]] auto getReplicatedLogById(replication2::LogId id) const
+      -> std::shared_ptr<replication2::replicated_log::ReplicatedLog const> {
+    if (auto iter = _logs.find(id); iter != _logs.end()) {
```

Comment:
Should this acquire the `_mutex` when accessing `_logs`?

Reply:
Fixed